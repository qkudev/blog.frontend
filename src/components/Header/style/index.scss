@import '../../../style/mixins/index';
@import '../../../style/variables';

.header {
  position: relative;
  display: grid;
  align-items: end;
  color: $white;
  background-color: $primary-darken;
  min-height: 100vh;
  max-height: 100vh;
  min-width: 100vw;
  max-width: 100vw;
  padding: 0 1rem;
  overflow: hidden;
  z-index: 5;

  a {
    padding-bottom: 1rem;
    font-size: 5rem;
    font-weight: 200;
    line-height: 5rem;
    color: $white;
    z-index: 150;
  }

  @media (max-width: 460px) {
    a {
      max-width: 258px;
    }
  }
  @media (min-width: 461px) and (max-width: 991px) {
    a {
      max-width: 430px;
    }
    padding: 0 1.5rem;
  }

  @media (min-width: 992px) {
    a {
      padding-bottom: 2rem;
    }
    padding: 0 3rem;
  }

  // Cyan image
  &::before {
    content: '\ ';
    position: absolute;
    left: -1rem;
    top: -1rem;
    right: -1rem;
    bottom: -1rem;
    min-height: 100vh;
    max-height: 2268px;
    max-width: 4032px;
    width: auto;

    background-image: url('header.jpg');
    -webkit-background-size: cover;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    background-color: rgba(0, 255, 255, 0.4);
    background-blend-mode: lighten;
    overflow: hidden;
    z-index: 100;
  }

  // Red image
  &::after {
    content: '\ ';
    position: absolute;
    left: -1rem;
    top: -1rem;
    right: -1rem;
    bottom: -1rem;
    min-height: 100vh;
    max-height: 2268px;
    max-width: 4032px;
    width: auto;

    background-image: url('header.jpg');
    -webkit-background-size: cover;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    background-color: #ff0000;
    background-blend-mode: darken;
    overflow: hidden;
    z-index: 105;
    opacity: 0.5;

    animation: show-overlap-left 1.5s ease-in-out 1.5s infinite alternate;

    // Animating with transform properties to use GPU
    // https://medium.com/ag-grid/javascript-gpu-animation-with-transform-and-translate-bf09c7000aa6
    @keyframes show-overlap-left {
      0% {
        transform: translateX(0);
      }
      100% {
        transform: translateX(-1em);
      }
    }
  }
}

#header-link {
}
