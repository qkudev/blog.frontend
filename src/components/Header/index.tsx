import * as React from 'react'
import { Link } from 'react-router-dom'
import './style'

function Header() {
  return (
    <div id='app-header' className='layout header'>
      <Link id='header-link' to='/'>
        Show me them all
      </Link>
    </div>
  )
}

export default Header
