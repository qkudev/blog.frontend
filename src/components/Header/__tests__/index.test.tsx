import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Header from '../index'

describe('Header', function() {
  test('Matches with snapshot', function() {
    const header = renderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    let tree = header.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
