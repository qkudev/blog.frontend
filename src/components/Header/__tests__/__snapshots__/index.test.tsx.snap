// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches with snapshot 1`] = `
<div
  className="layout header"
  id="app-header"
>
  <a
    href="/"
    id="header-link"
    onClick={[Function]}
  >
    Show me them all
  </a>
</div>
`;
